docs/ci-services.md

```diff
@@ -0,0 +1,102 @@
+# CI Services
+
+The CI services map defines the supported CI services the user will be
+presented with when running the scaffolder.
+
+## Example
+
+```javascript
+{
+  CIServiceName: {scaffolder: opts => scaffold(opts), public: true}
+}
+```
+
+## Existing scaffolders
+There are already multiple CI service scaffolders [available for
+consumption](https://github.com/form8ion/awesome#continuous-integration-services).
+Should your workflow fit into those, feel free to integrate them directly. If
+not, you are welcome and encouraged to build more (and maybe consider
+submitting a PR with a link 😎)!
+
+## Config
+
+### keys: __string__
+
+Keys in the CI services map define the choices the user will be presented with
+when choosing their desired CI service. These have no functional impact on the
+scaffolder.
+
+example: `CIServiceName`
+
+### values: __object__
+
+Values in the CI services map define the actual plugins and options that will
+be used by the JavaScript Scaffolder to scaffold the CI service for the user's
+project.
+
+#### scaffolder: __function__ (_required_)
+
+Consumer-provided plugin called when this CI service is selected by the
+user in the interactive prompt. Will be called with the following arguments:
+
+* `projectRoot` __string__ full path to the directory where scaffolder will
+  create the JavaScript project
+* `vcs` __object__
+  * `host` __string__ VCS hosting service to generate the repo with (e.g., `github`)
+  * `owner` __string__ account name on the host service for the repository to
+    generate (e.g., `form8ion`)
+  * `name` __string__ repository name to generate (e.g., `javascript-scaffolder`)
+* `visibility` __string__ visibility of the project to be scaffolded (valid
+  values are: `Public`, `Private`)
+* `projectType` __string__ type of project to be scaffolded (valid values are:
+  `Application`, `Package`, `CLI`)
+* `nodeVersionCategory` __string__ node category chosen for the project (valid
+  values are: `Latest`, `LTS`)
+* `tests` __object__
+  * `unit` __boolean__ whether the project will be unit-tested
+  * `integration` __boolean__ whether the project will be integration-tested
+
+*Response*
+
+The response object is commonly used across the project as a way for scaffolders
+to communicate information to the parent.
+
+* `nextSteps` __array__ list of strings that will be aggregated and displayed
```

Comment:
```suggestion
* `nextSteps` __array__ list of objects with shape of {summary: __string__, description: __markdown string__} that will be aggregated and displayed
```

not quite a suggestion of exact content, but this is the more correct type

Reply:
i got this shape from [here](https://github.com/travi/github-scaffolder/blob/9189718bd11a60dad5f2c7804ce94e895e82cd9e/src/next-steps.js#L13). am i understanding that one wrong?

Reply:
yeah, which might be pointing out confusing api design. the important piece to consider here is order of operations and apparently that multiple returned objects are using the same attribute name.

the results from sub-scaffolders, like the ci plugins, have [this shape](https://github.com/travi/github-scaffolder/blob/9189718bd11a60dad5f2c7804ce94e895e82cd9e/src/next-steps.js#L4), which are processed in that case to create github issues for each one. the version that you linked to is a list of urls that point to the created issues. keep in mind that the subscaffolders dont know which processors will read the results and that the github one is only one processor

Reply:
nice, good to know 👍 .  i went ahead and made the change, but i'm still confused. the docs we're shooting for in this change are the return values from the ci scaffolder. from what i'm seeing in the example you linked above it looks like the return value from the github scaffolder contains an object with `nextSteps` that is just a list of urls. 

this is making me think you might be talking about the values passed _in_ to the ci scaffolder, which i don't even have in the list of arguments to the top-level `scaffolder` on the ci-services object.

Reply:
oh, right... yeah, we probably need to improve the naming even more if i'm getting confused by this. you're right that i was getting the input/output wrong, but the reason for me getting it wrong is that i was pointing out the version of `nextSteps`	 that is more consistently returned from _general_ plugins. i was still thinking from the perspective of working toward documenting the general case and calling out how the specific cases differ from that. even calling out this difference is really confusing when they use the same name for very different purposes